packages/sdk-components-react/src/time.tsx

```diff
@@ -348,8 +348,21 @@ export const Time = React.forwardRef<React.ElementRef<"time">, TimeProps>(
       timeStyle: timeStyleOrUndefined(timeStyle),
     };
 
-    const datetimeString =
-      datetime === null ? INVALID_DATE_STRING : datetime.toString();
+    let datetimeString = INVALID_DATE_STRING;
+
+    if (datetime !== null) {
```

Comment:
Can't we normalize the timestamp first? seems much simpler
```js
let timestamp = Number(datetime);
if ( Number.isNaN(timestamp) === false && datetime.length === 10) {
  timestamp = timestamp * 1000
}

Reply:
turns out that 10 digit timestamp is the same unix timestamp but in seconds instead of milliseconds

Reply:
please also add a comment, I would have never guessed php is using this timestamp

Reply:
Following up from DM @kof 

> Can't we normalize the timestamp first? seems much simpler

If we only needed logic on the 10 digit yes, but we need logic for all timestamps. Converting to an iso string lets them get parsed again by the date parser. If we leave them as a timestamp string, they'll fail. So we'd need to cast all timestamps to numbers which the condition would grow again and have to change the logic elsewhere too.

My understanding is you wanted to do something like this:

```js
if (datetime) {
      let timestamp = Number(datetime);
      if (Number.isNaN(timestamp) === false && datetime.length === 10) {
        timestamp = timestamp * 1000
      } else {
        datetimeString = datetime.toString();
      }
    }
    ```
    
    But that leaves all other timestamps untouched
    
    Open to suggestions.

Reply:
at this point I am lost, lets get this into a function and cover it with unit tests, this is too confusing

Reply:
might be too much for you, so someone could take over, but need a list of inputs you are trying to support

Reply:
1698377386 - A valid 10-digit Unix timestamp (October 26, 2023).
2147483647 - The maximum 32-bit signed integer timestamp (January 19, 2038).
1698377386000 - A valid 13-digit Unix timestamp (October 26, 2023).
9999999999999 - A far future date (November 20, 2286).

Reply:
cool, that's def. not clear from reading code, needs unit tests, I can add them if you want me to take over

Reply:
that would be great, thanks

Reply:
Added tests, made sure they work with your implemeentation, then did what I suggested and tests passed, not sure what i am missing but all tests are passing